### ISSUE TITLE
Trainer: Functions to inspect model and optimizer status

### ISSUE THREAD

**Issue by yqy2001** [huggingface/transformers#29016]
### Feature request

In huggingface Trainer, are there any functions to inspect model and optimizer status? such as, how many parameters require grad, learning rate of each parameter, which optimizer group each parameter belong...

I didn't find any related function in Trainer, and I know implementing it by myself is easy, but I just want to know whether such functions already exist.

### Motivation

Such inspection is useful for correcting training.

### Your contribution

I propose a question.

**Comment by amyeroberts** [comment]
cc @muellerzr @pacman100

**Comment by CKeibel** [comment]
Hi, can I take over the issue?

**Comment by amyeroberts** [comment]
@CKeibel Sure! No need to claim on an issue, we prioritise based on PRs open, as we find this helps prevent issues from going stale without being addressed. Once you have something opened, feel free to ping me and @muellerzr for review 🤗

**Comment by CKeibel** [comment]
Hey, thanks for the reply. Maybe I should have read the contribution guidelines first. I'm used to asking beforehand in other open source projects. But then I will start contributing diligently. :)
